python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+                          rel_weight=rel_weight,
+                          footprint=fp,
+                          algObject=self)
+
+        if verbose:  # the ci_report() seems to fail if neg params are constrained -- TBD why.
+            # Never wanted in production - this takes a long time (longer than the fit!)
+            # This is how to get confidence intervals out:
+            #    https://lmfit.github.io/lmfit-py/confidence.html and
+            #    http://cars9.uchicago.edu/software/python/lmfit/model.html
+            print(result.fit_report(show_correl=False))
+            if separateNegParams:
+                print(result.ci_report())
+
+        # Display images, model fits and residuals (currently uses matplotlib display functions)
+        if display:
+            self.displayFitResults(fp, result)
+
+        return result
+
+    def fitDipole(self, source, tol=1e-7, rel_weight=0.1,
+                  fitBgGradient=True, maxSepInSigma=5., separateNegParams=True,
+                  bgGradientOrder=1, verbose=False, display=False, return_fitObj=False):
+        """Wrapper around `fitDipoleImpl()` which performs the fit of a dipole
+        model to an input difference image (actually, subimage bounded
+        by the input source's footprint) and optionally constrain the
+        fit using the pre-subtraction images self.posImage (science) and
+        self.negImage (template). Wraps the output into a `resultsOutput`
+        named tuple after computing additional statistics such as
+        orientation and SNR.
+
+        @param source Record containing the (merged) dipole source footprint detected on the diffim
+        @param tol Tolerance parameter for scipy.leastsq() optimization
+        @param rel_weight Weighting of posImage/negImage relative to the diffim in the fit
+        @param fitBgGradient Fit linear background gradient in posImage/negImage?
+        @param bgGradientOrder Desired polynomial order of background gradient (allowed are [0,1,2])
+        @param maxSepInSigma Allowed window of centroid parameters relative to peak in input source footprint
+        @param separateNegParams Fit separate parameters to the flux and background gradient in
+        the negative images?
+        If true, this adds a separate parameter for the negative flux, and [1, 3, or 6] additional
+        parameters to fit for the background gradient in the negImage. Otherwise, the flux and
+        gradient parameters are constrained
+        to be exactly equal in the fit.
+        @param verbose Be verbose
+        @param display Display input data, best fit model(s) and residuals in a matplotlib window.
+        @param return_fitObj In addition to the `resultsOutput` object, also return the
+        `lmfit.MinimizerResult` object for debugging.
+
+        @return resultsOutput object containing the fit parameters and other information.
+        @return lmfit.MinimizerResult object if `return_fitObj` is True.
```

Comment:
It the `MinimzerResult` is always computed anyway, so I can always return it. Done.